File: 프로그래머스/0/181888. n개 간격의 원소들/n개 간격의 원소들.py
def solution(num_list, n):
    
    return num_list[::n]

# 슬라이싱: 리스트[start:stop:step]
#    - start: 시작 위치(포함 O), 생략 시 처음부터
#    - stop: 종료 위치(포함 X), 생략 시 끝까지
#    - step: 건너뛸 간격, 생략시 1

# 다른 풀이
return num_list[0 : len(num_list) : n]

# 다른 풀이
return [num_list[i] for i in range(0, len(num_list), n)]
# range(start, stop, step): 슬라이싱의 인자 설명과 동일 

File: 프로그래머스/0/181888. n개 간격의 원소들/n개 간격의 원소들.java
class Solution {
    public int[] solution(int[] num_list, int n) {
        int size = 0;
        if(num_list.length % n == 0) 
            size = num_list.length / n;
        else 
            size = num_list.length / n + 1;
        
        int[] answer = new int[size];
        
        for(int i = 0, j = 0; i < num_list.length; i += n, j++){
            answer[j] = num_list[i];
        }
        
        return answer;
    }
}

// 다른 풀이
return IntStream.iterate(0, i -> i < numList.length, i -> i + n)
                .map(i -> numList[i])
                .toArray();
// .iterate() : 초기 값과 조건, 그리고 요소를 생성하는 함수를 인자로 받음
//    - 초기값 : 0
//    - 조건 : i -> i < numList.length
//    - 값 변경 함수 : i -> i + n
